client/src/components/MapView/LayerControl.tsx

```diff
@@ -85,15 +85,8 @@ const LayerControl: React.FC<LayerControlProps> = ({ mapRef, setInteractiveLayer
       mapInstance.setLayoutProperty(layerId, 'visibility', isVisible ? 'visible' : 'none');
 
       // After toggling, recalculate which interactive layers are visible.
-      const style = mapInstance.getStyle();
-      if (!style || !style.layers) return;
-
-      const visibleInteractive = style.layers
-        .filter((l) => l.type !== 'raster' && !l.id.startsWith('jsx'))
-        .filter((l) => mapInstance.getLayoutProperty(l.id, 'visibility') !== 'none')
-        .map((l) => l.id);
-
-      setInteractiveLayerIds(visibleInteractive);
+      const selected = findSelectedDebugLayers(mapInstance);
+      setInteractiveLayerIds(selected);
```

Comment:
There seems to be a change in behavior here. Previously, when `(!style || !style.layers`, the code path would never reach this point, but after this change, when the same condition is true, setInteractiveLayerIds is called with an empty array. Is this the intended change?

Reply:
Yes, in theory it's a change but for it to matter you have to open the right hand panel before the style.json has loaded. And even then I would argue that an empty list is the correct thing to return. Once the map has loaded the `interactiveLayerIds` are set to the correct value.